frontend/ui/widget/multiinputdialog.lua

```diff
@@ -110,6 +110,12 @@ function MultiInputDialog:init()
     self:onCloseKeyboard()
     self._input_widget:onCloseWidget()
 
+    -- Update self.keyboard_visible, this is needed because InputDialog:onCloseKeyboard updates it regardless of whether
+    -- G_reader_settings("virtual_keyboard_enabled") is disabled, which can lead to an incorrect keyboard visibility state.
```

Comment:
Can you find out why so? Maybe it should be fixed somewhere?

Reply:
That's what I meant by https://github.com/koreader/koreader/pull/12951#discussion_r1900437021

`InputDialog:onCloseKeyoard` does what it says on the tin: it closes the KB, which means it's no longer visible.

Modifying its behavior just to fix this very specific interaction seems like it could lead to more unintended consequences than just dealing with it this way here.

Reply:
> `InputDialog:onCloseKeyoard` does what it says on the tin: it closes the KB, which means it's no longer visible, so it forcibly always sets the visibility flag to `false`.

Well, technically, to `self._input_widget:isKeyboardVisible()`, which should pretty much always lead to a `false` after `InputText:onCloseKeyboard()` ;).